src/OpenFeature.Contrib.ConfigCat/ConfigCatProvider.cs

```diff
@@ -0,0 +1,97 @@
+using System.Threading.Tasks;
+using ConfigCat.Client;
+using ConfigCat.Client.Configuration;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+
+namespace OpenFeature.Contrib.ConfigCat
+{
+    /// <summary>
+    /// ConfigCatProvider is the .NET provider implementation for the feature flag solution ConfigCat.
+    /// </summary>
+    public class ConfigCatProvider : FeatureProvider
+    {
+        private const string Name = "ConfigCat Provider";
+        internal readonly IConfigCatClient Client;
+
+        /// <summary>
+        /// Creates new instance of <see cref="ConfigCatProvider"/>
+        /// </summary>
+        /// <param name="sdkKey">SDK Key to access the ConfigCat config.</param>
+        /// <param name="configBuilder">The action used to configure the client.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="sdkKey"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="sdkKey"/> is an empty string or in an invalid format.</exception>
+        public ConfigCatProvider(string sdkKey, Action<ConfigCatClientOptions> configBuilder = null)
+        {
+            Client = ConfigCatClient.Get(sdkKey, configBuilder);
+        }
+
+        /// <inheritdoc/>
+        public override Metadata GetMetadata()
+        {
+            return new Metadata(Name);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+        {
+            return await ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<string>> ResolveStringValue(string flagKey, string defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<int>> ResolveIntegerValue(string flagKey, int defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<double>> ResolveDoubleValue(string flagKey, double defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<ResolutionDetails<Value>> ResolveStructureValue(string flagKey, Value defaultValue, EvaluationContext context = null)
+        {
+            var stringDefaultValue = defaultValue?.AsString;
+            var result = await ProcessFlag(flagKey, context, stringDefaultValue);
+            var returnValue = result.Value == null ? defaultValue : new Value(result.Value);
+            return new ResolutionDetails<Value>(flagKey, returnValue, variant: result.Variant, errorMessage: result.ErrorMessage);

```

Comment:
This implementation looks problematic because this will work only when the feature flag value is a string (for more details, see https://configcat.com/docs/sdk-reference/dotnet/#setting-type-mapping).

However, `Value` can store values of other types than string (see [type definition](https://github.com/open-feature/dotnet-sdk/blob/e8ca1da9ed63df9685ec49a9569e0ec99ba0b3b9/src/OpenFeature/Model/Value.cs)).

To make this work with all the feature flag types which is supported by ConfigCat (boolean, string, int, double), we need something like this:

```suggestion
            var user = context?.BuildUser();
            var result = await Client.GetValueDetailsAsync(flagKey, defaultValue?.AsObject, user);
            var returnValue = result.IsDefaultValue ? defaultValue : new Value(result.Value);
            return new ResolutionDetails<Value>(flagKey, returnValue, ParseErrorType(result.ErrorMessage), errorMessage: result.ErrorMessage, variant: result.VariationId);
```